### PR TITLE
Fix some test failures

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

RED

### Solution

GREEN

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -97,13 +97,14 @@ jobs:
       - run: apk update && apk add git make curl tar
       - checkout
       - gomod
-      - setup_remote_docker
+      - setup_remote_docker:
```
Comment (AnnaShaleva) [inline .circleci/config.yml:100]: Our circleci `build_image` test is failing on this PR. Is this behaviour expected?

File: pkg/network/tcp_transport.go [hunk 2]
```diff
@@ -50,16 +51,23 @@ func (t *TCPTransport) Accept() {
 	}
 
 	t.lock.Lock()
+	if t.quit {
+		l.Close()
+		return
```
Comment (AnnaShaleva) [inline pkg/network/tcp_transport.go:56]: Don't need `t.lock.Unlock` before it?

File: .circleci/config.yml (hunk 1 above)
Comment (roman-khimov) [reply .circleci/config.yml:100]: It actually worked for like 5 previous runs, but failed this time. Now I don't know how to fix it. This was supposed to work.

File: pkg/network/tcp_transport.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/network/tcp_transport.go:57]: Sure, fixed.

**Comment by roman-khimov** [comment]
Discovered is fixed too now. Let's go :green_circle:

**Review by fyrchik — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]
Nice!

**Pull request merged**